File: Assignment4/Program5.c
//5.Write a program which accept number from user and return difference between summation
//of all its factors and non factors.
//Input :  12 Output : -34  (16 - 50)  //Input :  10  Output : -29 (8 - 37)
#include <stdio.h>
int FactDiff(int iNo)
{ 
    int iCnt = 0; int iSum1 = 0;int iSum2 = 0; int diff = 0;
    if (iNo < 0)
    {
        iNo = -iNo;
    }
    for (iCnt = 1; iCnt < iNo; iCnt++)
    {
        if (iNo % iCnt == 0)
        {
            iSum1 = iSum1 + iCnt;
        }
        else { iSum2 = iSum2 + iCnt; }
        diff = iSum1 - iSum2;
    }
    return diff;
}
int main()
{
    int iValue = 0;
    int iRet = 0;
    printf("Enter number");
    scanf("%d", &iValue);
    iRet = FactDiff(iValue);
    printf("%d", iRet);
    return 0;
}

File: Assignment33/Program5.java
// 5.Write a program which accept number from user and return difference between summation 
// of even digits and 
// summation of odd digits. 
// Input :  2395  Output :  -15 (2 - 17) 
// Input :  1018 Output :  6 (8 - 2) 
// Input :  8440 Output :  16  (16 - 0) 
// Input :  5733 Output :  -18 (0 - 18) 

import java.util.*;

class Program5
{
	public static void main(String[] args) 
	{
		Scanner sobj = new Scanner(System.in);
		int iValue = 0,iRet = 0;

		System.out.println("Enter number");
		iValue = sobj.nextInt();

		Digit dobj =new Digit();
		iRet =dobj.CountDiff(iValue);

		System.out.println("Difference is : "+iRet);
	}
}

class Digit 
{  
	public int CountDiff(int iNo)  
	{   
		int iDigit = 0,iSumEven = 0,iSumOdd = 0,iDiff = 0;

		while(iNo != 0)
		{
			iDigit = iNo % 10;
			iNo = iNo / 10;
			if(iDigit % 2 == 0)
			{
				iSumEven = iSumEven + iDigit;
			}
			else
			{
				iSumOdd = iSumOdd + iDigit;
			}
			iDiff = iSumEven - iSumOdd;
		}
		return iDiff;
	} 
}
